sympy/stats/rv.py

```diff
@@ -1428,6 +1429,49 @@ def __getattr__(self, attr):
             raise AttributeError("'%s' object has no attribute '%s'" % (
                 type(self).__name__, attr))
 
+
+class Distribution(Basic):
+
+    def sample(self, size=(), library='scipy'):
+        """ A random realization from the distribution """
+
+        module = import_module(library)
+        if library in {'scipy', 'numpy', 'pymc3'} and module is None:
+            raise ValueError("Failed to import %s" % library)
+
+        if library == 'scipy':
+            # scipy does not require map as it can handle using custom distributions.
+            # However, we will still use a map where we can.
+
+            # TODO: do this for drv.py and frv.py if necessary.
+            # TODO: add more distributions here if there are more
+            # See links below referring to sections beginning with "A common parametrization..."
+            # I will remove all these comments if everything is ok.
+
+            from sympy.stats.sampling.sample_scipy import do_sample_scipy
+            samps = do_sample_scipy(self, size)
+        elif library == 'numpy':
+            from sympy.stats.sampling.sample_numpy import do_sample_numpy
+            samps = do_sample_numpy(self, size)
+        elif library == 'pymc3':
+            from sympy.stats.sampling.sample_pymc3 import do_sample_pymc3
+            import pymc3
+            with pymc3.Model():
+                if do_sample_pymc3(self):
+                    samps = pymc3.sample(size, chains=1, progressbar=False)[:]['X']
+                else:
+                    samps = None
+        else:
+            raise NotImplementedError("Sampling from %s is not supported yet."
+                                      % str(library))
+
+        if samps is not None:
+            return samps
+        raise NotImplementedError(
+            "Sampling for %s is not currently implemented from %s"
+            % (self.__class__.__name__, library))
```

Comment:
I think using `self` should be okay here. We should give exactly the object which caused the error instead of the class so that it's easy to debug for the end user.
In addition, we will be able to avoid usage of `self.__class__.__name__`.